open-api/rest-catalog-open-api.yaml

```diff
@@ -3789,6 +4003,21 @@ components:
             EmptyResponse:
               $ref: '#/components/examples/ListNamespacesEmptyExample'
 
+    PayloadTooLargeWithPlanResponse:
+      description:
+        Payload is too large. Required to use /preplan before /plan
+      content:
+        application/json:
+          schema:
+            $ref: '#/components/schemas/IcebergErrorResponse'
+          example: {
+            "error": {
+              "message": " Cannot plan scan: too many task results; use /preplan to split planning across multiple requests",
+              "type": "PayloadTooLargeWithPlanException",
+              "code": 413
```

Comment:
I think `413` isn't the right code here.  This code indicates the request content is too large, not the response.  I believe `422` is more appropriate

Reply:
I suggested either 413 or 422, but you're right. 422 is the better option.